test/new-e2e/pkg/e2e/suite.go

```diff
@@ -222,12 +222,19 @@ func (bs *BaseSuite[Env]) IsDevMode() bool {
 	return bs.params.devMode
 }
 
-func (bs *BaseSuite[Env]) init(options []SuiteOption, self Suite[Env]) {
+func (bs *BaseSuite[Env]) init(t *testing.T, options []SuiteOption, self Suite[Env]) {
+	devMode, err := runner.GetProfile().ParamStore().GetBoolWithDefault(parameters.DevMode, false)
+	if err != nil {
+		t.Logf("dev_mode parameter error : %s", err)
+	} else if devMode {
+		options = append(options, WithDevMode())
+	}
+
 	for _, o := range options {
 		o(&bs.params)
 	}
 
-	if bs.params.devMode && !runner.GetProfile().AllowDevMode() {
+	if !runner.GetProfile().AllowDevMode() {
 		bs.params.devMode = false
 	}
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
```suggestion
    // disable dev mode if dev mode is not allowed
	bs.params.devMode = bs.params.devMode && runner.GetProfile().AllowDevMode()
```
Feel free to discard, not a blocker and your code might be more readable

Reply:
I like oneliner :) but all the gocode is if {}, let's keep easy for new reader